ping.go

```diff
@@ -84,10 +84,20 @@ func ping(w http.ResponseWriter, r *http.Request) {
 		ip = r.RemoteAddr
 	}
 
+	var ua string
+	if res := r.Header.Get("User-Agent"); res != "" {
+		ua = res
+	} else {
+		log.Println("empty user-agent")
+		javascriptRespond(w, http.StatusBadRequest, "empty user-agent")
+		return
+	}
```

Comment:
I think I would refactor this into:

```go
userAgent := r.Header.Get("User-Agent")
if userAgent == "" {
    log.Println("empty user-agent")
    javascriptRespond(w, http.StatusBadRequest, "empty user-agent")
    return
}
```

I think it's a bit more direct that way. What do you think?